fdbclient/NativeAPI.actor.cpp

```diff
@@ -8296,21 +8297,37 @@ ACTOR static Future<int64_t> rebootWorkerActor(DatabaseContext* cx, ValueRef add
 	state std::map<Key, std::pair<Value, ClientLeaderRegInterface>> address_interface;
 	if (!cx->getConnectionRecord())
 		return 0;
+	// split and get all the addresses to send reboot request
+	state std::vector<std::string> addressesVec;
+	boost::algorithm::split(addressesVec, addr.toString(), boost::is_any_of(","));
 	RangeResult kvs = wait(getWorkerInterfaces(cx->getConnectionRecord()));
 	ASSERT(!kvs.more);
 	// Note: reuse this knob from fdbcli, change it if necessary
 	Reference<FlowLock> connectLock(new FlowLock(CLIENT_KNOBS->CLI_CONNECT_PARALLELISM));
 	std::vector<Future<Void>> addInterfs;
+	std::unordered_set<std::string> addressesSet(addressesVec.begin(), addressesVec.end());
 	for (const auto& it : kvs) {
-		addInterfs.push_back(addInterfaceActor(&address_interface, connectLock, it));
+		KeyRef address =
+		    it.key.endsWith(LiteralStringRef(":tls")) ? it.key.removeSuffix(LiteralStringRef(":tls")) : it.key;
+		// only fetch requested workers' interfaces
+		if (addressesSet.count(address.toString())) {
+			addInterfs.push_back(addInterfaceActor(&address_interface, connectLock, it));
+		}
 	}
 	wait(waitForAll(addInterfs));
-	if (!address_interface.count(addr))
-		return 0;
 
-	BinaryReader::fromStringRef<ClientWorkerInterface>(address_interface[addr].first, IncludeVersion())
-	    .reboot.send(RebootRequest(false, check, duration));
-	return 1;
+	int64_t result = 1;
+	for (const auto& address : addressesVec) {
+		if (address_interface.count(ValueRef(address))) {
+			// Note: We want to make sure these requests are sent in parallel
+			BinaryReader::fromStringRef<ClientWorkerInterface>(address_interface[ValueRef(address)].first,
```

Comment:
@jzhou77 here is where the processes are actually killed. The CLI_CONNECT_PARALLELISM knob is for establishing connections first

Reply:
yeah, here is where we send out the reboot requests.

Reply:
Yes, I understand now.